src/chain/mod.rs

```diff
@@ -113,149 +458,545 @@ async fn get_nonce(port: u16, client: &Client, address: &str) -> Result<u64> {
+    for contract in &config.contracts {
+        if let (Some(name), Some(address)) = (&contract.name, &contract.address) {
+            deployed_addresses.insert(name.clone(), address.clone());
+            info!("Pre-registered contract '{}' at {}", name, address);
+        }
+    }
+
+    let _impersonator = AnvilImpersonator::new(port, &client, OWNER_ADDRESS).await?;
+    let mut nonce = get_nonce(port, &client, OWNER_ADDRESS).await?;
+
+    // Deploy contracts sequentially
+    for contract in &config.contracts {
+        if let Some(json_path) = &contract.contract_json_path {
+            let name = contract.name.as_deref().unwrap_or("unnamed");
+
+            let mut bytecode = load_creation_bytecode(json_path)?;
+
+            // Append constructor args if any
+            if !contract.constructor_args.is_empty() {
+                let encoded_args =
+                    encode_constructor_args(&contract.constructor_args, &deployed_addresses)?;
+                bytecode = format!(
+                    "0x{}{}",
+                    bytecode.trim_start_matches("0x"),
+                    encoded_args.trim_start_matches("0x")
+                );
+            }
+
+            info!(
+                "Deploying contract '{}' with bytecode length: {}",
+                name,
+                bytecode.len()
+            );
+
+            match execute_transaction(port, &client, OWNER_ADDRESS, None, &bytecode, nonce).await {
+                Ok(tx_hash) => {
+                    info!("Deployment tx for '{}': {}", name, tx_hash);
+
+                    // Wait for receipt
+                    for _ in 0..10 {
+                        sleep(Duration::from_millis(100)).await;
+                        if let Ok(Some(contract_address)) =
+                            get_transaction_receipt(port, &client, &tx_hash).await
+                        {
+                            info!("Contract '{}' deployed at: {}", name, contract_address);
+                            if let Some(name) = &contract.name {
+                                deployed_addresses.insert(name.clone(), contract_address);
+                            }
+                            break;
+                        }
+                    }
+                }
+                Err(e) => {
+                    info!("Failed to deploy contract '{}': {}", name, e);
+                }
+            }
+
+            nonce += 1;
+        }
+    }
+
+    Ok(deployed_addresses)
+}
+
+#[instrument(level = "trace", skip_all)]
+async fn execute_config_transactions(
+    port: u16,
+    config: &ChainConfig,
+    deployed: &HashMap<String, String>,
+) -> Result<()> {
+    if config.transactions.is_empty() {
+        return Ok(());
+    }
+
+    info!(
+        "Found {} configured transactions",
+        config.transactions.len()
+    );
+
+    let client = Client::new();
+    let _impersonator = AnvilImpersonator::new(port, &client, OWNER_ADDRESS).await?;
```

Comment:
do we actually need to impersonate here since we execute_transaction with OWNER_ADDRESS as from in [L709](https://github.com/hyperware-ai/kit/blob/9c9973f739422ed0a4652d65cc549480808356c4/src/chain/mod.rs#L709)?